### ISSUE TITLE
A more natural way to create unbounded Intervals

### ISSUE THREAD

**Issue by haraldk** [ThreeTen/threeten-extra#184]
Currently, the way to create an unbounded Interval, is to use the `Interval.of(Instant, Instant)` factory method with the start/end explicitly defined as `Instant.MIN/Instant.MAX`. Or use `Interval.ALL.withStart()/withEnd()` which is, perhaps,  slightly better. But none of these feel natural to me.

I'm proposing a new factory method (or two) in `Interval` for this. I'm not entirely sure of what the names should be though, but something like this:

````java
    static Interval ofStart(Instant start) {
        return ALL.withStart(start);
    }

    static Interval.ofEnd(Instant end) {
        return ALL.withEnd(end);
    }
````

PS: I often find myself in need of an empty "none" or "never" interval, to use instead of `null`, so it would also be nice to have that as a constant in `Interval` too. For this I'm currently using `Interval.of(Instant.MAX, Duration.ZERO)`. 

If this sounds like a good idea, I can probably create a PR for this.

**Comment by perceptron8** [comment]
+1 for `ofStart` / `ofEnd`.

Not quite sure about "null" interval as there is no single, obvious candidate. Of course, as always, it's up to @jodastephen to decide.

BTW because of #100, you can't create `LocalDateRange.of(LocalDate.MAX, Period.ofDays(0))` and it looks like you shouldn't be able to create `Interval.of(Instant.MAX, Duration.ZERO)` too for the same reasons.

```
Interval zero = Interval.of(Instant.MAX, Duration.ZERO);
assertTrue(zero.isEmpty());
assertFalse(zero.isUnboundedEnd()); // FAIL
assertFalse(zero.contains(Instant.MAX)); // FAIL
```

... or there should be real distinction between unbounded and starting/ending at MIN/MAX.

**Comment by haraldk** [comment]
There's absolutely some interesting semantics there... 😉  

The API docs for `Interval.contains` says:

> If this range has an unbounded end then `contains(Instant#MAX)` returns `true`. If this range is empty then this method always returns `false`.

My interval has both an unbounded end *and* is empty... So it's a bit unclear, but it's obviously not *always* false. Although I think it would make sense if the last rule took precedence, unless I'm missing some cases here. 

Whether or not "never" is unbounded is mostly philosophical to me.

**Comment by jodastephen** [comment]
Maybe `startingAt` and `endingAt`? I don't see how we could pick values for a "null" object.

It might be possible to change the semantics of `Interval` in a v2.0, but only in very minor ways. Perhaps an alternative to `getStart` and `getEnd` that return `Optional<Instant>` instead of `Instant` would be a step forward.

**Comment by jodastephen** [comment]
Fixed by #187
